core/drivers/stm32_fmc.c

```diff
@@ -244,28 +246,33 @@ static TEE_Result parse_dt(const void *fdt, int node)
 		return res;
 
 	cuint = fdt_getprop(fdt, node, "st,protreg", &lenp);
-	if (!cuint)
-		panic("No RIF configuration available");
+	if (!cuint) {
+		DMSG("No RIF configuration available");
+		goto skip_rif;
+	}
+
+	fmc_d->conf_data = calloc(1, sizeof(*fmc_d->conf_data));
+	if (!fmc_d->conf_data)
+		panic();
 
 	fmc_d->nb_controller = (unsigned int)(lenp / sizeof(uint32_t));
 	assert(fmc_d->nb_controller <= FMC_RIF_CONTROLLERS);
 
-	fmc_d->conf_data.cid_confs = calloc(FMC_RIF_CONTROLLERS,
-					    sizeof(uint32_t));
-	fmc_d->conf_data.sec_conf = calloc(1, sizeof(uint32_t));
-	fmc_d->conf_data.priv_conf = calloc(1, sizeof(uint32_t));
-	fmc_d->conf_data.lock_conf = calloc(1, sizeof(uint32_t));
-	fmc_d->conf_data.access_mask = calloc(1, sizeof(uint32_t));
-	assert(fmc_d->conf_data.cid_confs && fmc_d->conf_data.sec_conf &&
-	       fmc_d->conf_data.priv_conf && fmc_d->conf_data.access_mask);
+	fmc_d->conf_data->cid_confs = calloc(FMC_RIF_CONTROLLERS,
+					     sizeof(uint32_t));
+	fmc_d->conf_data->sec_conf = calloc(1, sizeof(uint32_t));
+	fmc_d->conf_data->priv_conf = calloc(1, sizeof(uint32_t));
+	fmc_d->conf_data->lock_conf = calloc(1, sizeof(uint32_t));
+	fmc_d->conf_data->access_mask = calloc(1, sizeof(uint32_t));
+	if (!fmc_d->conf_data->cid_confs || !fmc_d->conf_data->sec_conf ||
+	    !fmc_d->conf_data->priv_conf || !fmc_d->conf_data->access_mask)
+		panic("Missing memory capacity for FMC RIF configuration");
 
-	for (i = 0; i < fmc_d->nb_controller; i++) {
-		rif_conf = fdt32_to_cpu(cuint[i]);
+	stm32_rif_parse_cfg(fdt32_to_cpu(cuint[i]), fmc_d->conf_data,
```

Comment:
You forgot to restore the `for()` loop in commit "drivers: stm32_fmc: do not make RIF configuration mandatory".